src/core/nixl_metadata_manager.cpp

```
@@ -0,0 +1,242 @@
+
+    try {
+        if (worker_exception_) {
+            std::rethrow_exception(worker_exception_);
+        }
+    }
+    catch (const std::exception &e) {
+        NIXL_WARN << "Communication thread has thrown an exception: " << e.what();
+    }
+}
+
+void
```

Cppcheck (2.20.0)

[style] The function 'addErrorTelemetry' is never used.

(unusedFunction)